python/tvm/relay/backend/contrib/ethosu/te/dma.py

```diff
@@ -0,0 +1,299 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name,unnecessary-lambda
+"""Tensor Expressions for operations supported by the DMA engine"""
```

Comment:
```suggestion
"""Tensor Expressions for operations supported by the Ethos-U DMA engine."""
```

Reply:
Can you maybe explain why all the operations need to be redefined here? I think it is due to the fact that Ethos-U code generator matches its own TE for lowering, but its hard to tell from the code.

Reply:
Yeah you're pretty much spot on, the TE we create is very closely coupled to the functionality of our NPU and as such we don't want to rely on existing definitions which may be changed. We need it to remain identical so the TIR matching works and we attach some extra information to the TE (usually in the form of attributes) to ease that process.

Reply:
I've updated the comment here, but has used NPU in preference to Ethos-U.